src/ducks/Watchlists/Widgets/TopBar/TopBar.js

```diff
@@ -49,25 +70,61 @@ const TopBar = ({
   const [isCommentsOpen, setIsCommentsOpen] = useState(false)
   const currentEntity = getCurrentEntity({ entity, type })
   const { user: currentUser, isLoggedIn } = useUser()
-  const { user, name: title, description } = currentEntity
+  const {
+    user,
+    name: title,
+    id,
+    description,
+    commentsCount,
+    votes,
+    isPublic
+  } = currentEntity
   const { data } = usePublicUserData({ userId: user.id })
   const { isAuthor, isAuthorLoading } = useIsAuthor(entity)
   const [isFilterOpen, setIsFilterOpen] = useState(false)
+  const [updateWatchlist, { loading }] = useUpdateWatchlist(type)
+  const [isEditFormOpened, setIsEditFormOpened] = useState(false)
 
-  const {
-    avatarUrl,
-    username: userName,
-    name: fullName,
-    watchlists,
-    addressesWatchlists
-  } = data || {}
+  const { username: userName, id: userId } = data || {}
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      const comment = lookupSavedComment()
+
+      if (comment) {
+        setIsCommentsOpen(true)
+      }
+    }
+  }, [id])
+
+  function onVote () {
+    track.event(Event.LikeWatchlist || 'LikeWatchlist', { id })
+  }
 
   function closeFilter () {
     if (isFilterOpen) {
       setIsFilterOpen(false)
     }
   }
 
+  function handleSavedWatchlistComment () {
+    const node = document.querySelector(`textarea[name="comment"]`)
+    if (node) {
+      const comment = lookupSavedComment()
+      if (comment) {
+        node.value = comment.content
+        clearSavedComment()
+      }
+    }
+  }
+
+  const onEditApprove = props =>
+    updateWatchlist(entity, { ...props }).then(() => {
+      setIsEditFormOpened(false)
+      notifyUpdate(title)
+      refetchAssets && refetchAssets()
```

Comment:
For better readability lines should contain statements, i.e use `if (refetchAssets) ...`